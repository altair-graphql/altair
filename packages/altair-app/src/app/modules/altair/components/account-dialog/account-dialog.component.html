<app-dialog
  [showDialog]="showDialog"
  [heading]="'ACCOUNT_TITLE' | translate"
  [subheading]="'ACCOUNT_SUB_TEXT' | translate"
  [showFooter]="false"
  (toggleDialog)="toggleDialogChange.emit($event)"
>
<div class="account-signin-wrapper">
  <ng-container
    *ngIf="!account?.loggedIn"
  >
    <p>Sign in to manage your teams, synced collections and queries.</p>
    <button
      class="app-button active-primary"
      (click)="submitLogin()"
    >
      {{ 'SIGNIN_WITH_GOOGLE' | translate }}
    </button>
  </ng-container>

  <ng-container
    *ngIf="account?.loggedIn"
  >
    <div class="app-dialog-section">
      <div class="account-dialog__welcome">
        <nz-avatar [nzText]="account?.firstName?.at(0)" [nzSrc]="account?.picture"></nz-avatar>
        {{ 'ACCOUNT_HELLO' | translate }}, {{ account?.firstName }}
      </div>
    </div>
    <p>
      <a class="link" target="_blank" [href]="dashboardUrl">
        Dashboard
        <app-icon name="external-link"></app-icon>
      </a>
    </p>
    <div class="account-dialog__stats">
      <div class="account-dialog__stats-title">
        Plan Stats <app-icon name="info" [popper]="'The stats show how much resources you have and the number of resources allowed for your current plan'"></app-icon>
      </div>
      <div>Queries: {{ account?.stats?.queriesCount ?? 0 }} / {{ account?.plan?.maxQueriesCount ?? 0 }}</div>
      <div>Teams: {{ account?.stats?.teamsCount ?? 0 }} / {{ account?.plan?.maxTeamsCount ?? 0 }}</div>
    </div>
    <button
      class="app-button active-grey"
      (click)="logoutChange.emit()"
    >
      {{ 'LOGOUT_TEXT' | translate }}
    </button>
  </ng-container>
</div>
</app-dialog>
