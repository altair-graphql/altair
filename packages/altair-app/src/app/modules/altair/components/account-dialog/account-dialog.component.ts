import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { AccountState } from 'altair-graphql-core/build/types/state/account.interfaces';
import { environment } from 'environments/environment';
import { apiClient } from '../../services/api/api.service';
import { externalLink } from '../../utils';

@Component({
  selector: 'app-account-dialog',
  templateUrl: './account-dialog.component.html',
  styles: [],
})
export class AccountDialogComponent {
  @Input() showDialog = true;
  @Input() account: AccountState | undefined;
  @Output() toggleDialogChange = new EventEmitter<boolean>();
  @Output() handleLoginChange = new EventEmitter();
  @Output() logoutChange = new EventEmitter();

  dashboardUrl = apiClient.options.dashboardUrl;

  submitLogin() {
    this.handleLoginChange.emit();
  }
  async openBillingPage(e: MouseEvent) {
    const { url } = await apiClient.getBillingUrl();
    return externalLink(e, url);
  }
}
