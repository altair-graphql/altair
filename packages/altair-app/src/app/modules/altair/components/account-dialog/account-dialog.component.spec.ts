import { ComponentFixture, TestBed } from '@angular/core/testing';

import { AccountDialogComponent } from './account-dialog.component';

describe('AccountDialogComponent', () => {
  let component: AccountDialogComponent;
  let fixture: ComponentFixture<AccountDialogComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ AccountDialogComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AccountDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
