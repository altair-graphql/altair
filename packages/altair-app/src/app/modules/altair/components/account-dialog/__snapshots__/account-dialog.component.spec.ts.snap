// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDialogComponent should create 1`] = `
<app-account-dialog>
  <app-dialog>
    <div
      class="account-signin-wrapper"
    >
      <p>
        Sign in to manage your teams, synced collections and queries.
      </p>
      <button
        class="btn btn--primary"
      >
         SIGNIN_WITH_GOOGLE 
      </button>
      
      
    </div>
  </app-dialog>
</app-account-dialog>
`;
